search2/python/search/save/handler.py

```diff
@@ -1,43 +1,26 @@
+"""A module to handle the saving of search results into a new annotation"""
+# TODO 2023-05-08: Track number of skipped entries
+# TODO 2023-05-08: Write to Arrow IPC or Parquet, alongside tsv.
+# TODO 2023-05-08: Implement distributed pipeline/filters/transforms
+# TODO 2023-05-08: Support sort queries
+# TODO 2023-05-08: get max_slices from opensearch index settings
+# TODO 2023-05-08: concatenate chunks in a different ray worker
+
 import math
 import os
 import pathlib
 import subprocess
 import traceback
 
-import mgzip
+import mgzip # type: ignore
 import numpy as np
 from opensearchpy import OpenSearch
-from orjson import dumps
-from pystalk import BeanstalkClient
 import ray
 
-ray.init(ignore_reinit_error='true', address='auto')
-
-# TODO: track skipped
-# TODO: handle 1) the munging of the data, 2) distributed pipeline/transform , 3) write as arrow table (arrow ipc format), csv
-
-@ray.remote(num_cpus=0)
-class ProgressReporter:
-    def __init__(self, publisher: dict):
-        self.value = 0
-        self.publisher = publisher
-        self.client = BeanstalkClient(
-            publisher['host'], publisher['port'], socket_timeout=10)
-
-    def increment(self, count: int):
-        self.value += count
-        self.publisher['messageBase']['data'] = {
-            "progress": self.value,
-            "skipped": 0
-        }
-        self.client.put_job_into(
-            self.publisher['queue'], dumps(self.publisher['messageBase']))
-
-        return self.value
-
-    def get_counter(self):
-        return self.value
+from search.utils.beanstalkd import Publisher
+from search.index.handler import ProgressReporter
 
+ray.init(ignore_reinit_error='true', address='auto')
 
 default_delimiters = {
```

Comment:
nit: module-level constants should be UPCASED, and if a dict should be immutable you can lock it down with `frozendict`